src/workerd/api/tests/js-rpc-test.js

```diff
@@ -1695,7 +1697,9 @@ export let testAsyncStackTrace = {
     try {
       await env.MyService.throwingMethod();
     } catch (e) {
-      // verify stack trace was produced
+      // check that the custom property made it through
+      assert.strictEqual(e.abc, 123);
+      // verify a local stack trace was produced
       assert.strictEqual(e.stack.includes('at async Object.test'), true);
```

Comment:
Adding the abc 123 bit here demonstrates that the enhanced handling is working, and the unmodified stack check demonstrates that we're seeing local stacks, not the remote stack.